pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -111,37 +111,37 @@ fi
 
 if [[ "$isCpp" = 1 ]]; then
     if [[ "$cppInclude" = 1 ]]; then
-        NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE ${NIX_CXXSTDLIB_COMPILE-@default_cxx_stdlib_compile@}"
+        NIX_@infixSalt@_CFLAGS_COMPILE="$NIX_@infixSalt@_CFLAGS_COMPILE ${NIX_@infixSalt@_CXXSTDLIB_COMPILE-@default_cxx_stdlib_compile@}"
     fi
-    NIX_CFLAGS_LINK="$NIX_CFLAGS_LINK $NIX_CXXSTDLIB_LINK"
+    NIX_@infixSalt@_CFLAGS_LINK="$NIX_@infixSalt@_CFLAGS_LINK $NIX_@infixSalt@_CXXSTDLIB_LINK"
 fi
 
 LD=@ldPath@/ld
 source @out@/nix-support/add-hardening.sh
```

Comment:
@0xABAB good to know, but this `infixSalt` is something the derivation itself controls, so it's guaranteed to not break the identifier. Shouldn't that make it OK then?